x-pack/plugin/old-lucene-versions/src/main/java/org/elasticsearch/xpack/lucene/bwc/codecs/lucene70/BWCLucene70Codec.java

```diff
@@ -60,7 +49,18 @@ public BWCLucene70Codec() {
 
     protected BWCLucene70Codec(String name) {
         super(name);
-        storedFieldsFormat = new Lucene50StoredFieldsFormat(Lucene50StoredFieldsFormat.Mode.BEST_SPEED);
+        this.fieldInfosFormat = wrap(new Lucene60FieldInfosFormat());
+        this.segmentInfosFormat = wrap(new Lucene70SegmentInfoFormat());
+        this.storedFieldsFormat = new Lucene50StoredFieldsFormat(Lucene50StoredFieldsFormat.Mode.BEST_SPEED);
+        this.liveDocsFormat = new Lucene50LiveDocsFormat();
+        this.compoundFormat = new Lucene50CompoundFormat();
+        this.docValuesFormat = new PerFieldDocValuesFormat() {
+            @Override
+            public DocValuesFormat getDocValuesFormatForField(String field) {
+                return new Lucene80DocValuesFormat();
```

Comment:
I think that it was better before, because we were instantiating the format once instead of every time get was called for a certain field.